libs/ledger-live-common/src/deviceSDK/actions/updateFirmware.ts

```diff
@@ -0,0 +1,107 @@
+import { DeviceId, DeviceInfo } from "@ledgerhq/types-live";
+import { Observable, of } from "rxjs";
+import { filter, scan, switchMap } from "rxjs/operators";
+import {
+  updateFirmwareTask,
+  UpdateFirmwareTaskEvent,
+} from "../tasks/updateFirmware";
+import {
+  getDeviceInfoTask,
+  GetDeviceInfoTaskEvent,
+} from "../tasks/getDeviceInfo";
+import {
+  FullActionState,
+  initialSharedActionState,
+  sharedReducer,
+} from "./core";
+import {
+  getLatestFirmwareTask,
+  GetLatestFirmwareTaskErrorEvent,
+} from "../tasks/getLatestFirmware";
+
+export type updateFirmwareActionArgs = {
+  deviceId: DeviceId;
+};
+
+export type UpdateFirmwareActionState = FullActionState<{
+  step:
+    | "installingOsu"
+    | "flashingMcu"
+    | "flashingBootloader"
+    | "installOsuDevicePermissionRequested"
+    | "installOsuDevicePermissionGranted"
+    | "allowSecureChannelRequested"
+    | "firmwareUpdateCompleted"
+    | "preparingUpdate";
+  progress: number;
+  error: { type: "UpdateFirmwareError"; message?: string };
+}>;
+
+export const initialState: UpdateFirmwareActionState = {
+  step: "preparingUpdate",
+  progress: 0,
+  ...initialSharedActionState,
+};
+
+export function updateFirmwareAction({
```

Comment:
Maybe let's add a "JSDoc" documentation for each action, as they will be provided to our consumers

Reply:
Also with a small definition of what each `step` represents, so the consumer knows what to render

Reply:
done